### PR TITLE
Implement a new KeyStore file to configure the persistence logic

### PR DESCRIPTION
**Pull request by mcasas993**
### Description
This PR implements the configuration persistence method, so the Command Manager plugin can send data to the Management API in the server. With this, it will know the endpoint of its HTTP Rest API and its credentials in order to establish the communication.

### Issues Resolved
https://github.com/wazuh/wazuh-indexer-plugins/issues/95

### CODE REVIEW

File: plugins/command-manager/src/main/java/com/wazuh/commandmanager/settings/PluginSettings.java [hunk 1]
```diff
@@ -0,0 +1,88 @@
...
+    private static final String KEYSTORE_FILENAME = "wazuh-indexer.keystore";
+
+    private static KeyStoreWrapper keyStoreWrapper = KeyStoreWrapper.create();
+    private static Environment env;
+
+    private PluginSettings() {
+        // Singleton class, use getPluginSettings method instead of constructor
+    }
+
+    public static PluginSettings getPluginSettingsInstance() {
```
Comment (AlexRuiz7) [inline plugins/command-manager/src/main/java/com/wazuh/commandmanager/settings/PluginSettings.java:35]: Simplify to `getInstance()`

File: plugins/command-manager/src/main/java/com/wazuh/commandmanager/settings/PluginSettings.java [hunk 2]
```diff
@@ -0,0 +1,88 @@
...
+        if (INSTANCE != null) {
+            return INSTANCE;
+        }
+        synchronized (PluginSettings.class) {
+            if (INSTANCE != null) {
+                return INSTANCE;
+            }
```
Comment (AlexRuiz7) [inline plugins/command-manager/src/main/java/com/wazuh/commandmanager/settings/PluginSettings.java:42]: Why is this block duplicated?

File: plugins/command-manager/src/main/java/com/wazuh/commandmanager/settings/PluginSettings.java [hunk 3]
```diff
@@ -0,0 +1,88 @@
...
+            INSTANCE = new PluginSettings();
+            return INSTANCE;
+        }
+    }
+
+    public void setEnv(Environment env) {
```
Comment (AlexRuiz7) [inline plugins/command-manager/src/main/java/com/wazuh/commandmanager/settings/PluginSettings.java:48]: Use complete words and names. Rename to `setEnvironment`
The same applies for the static class member `PluginSettings.environment`.

File: plugins/command-manager/src/main/java/com/wazuh/commandmanager/settings/PluginSettings.java [hunk 4]
```diff
@@ -0,0 +1,88 @@
...
+
+public class PluginSettings {
+    private static final Logger logger = (Logger) LogManager.getLogger(PluginSettings.class);
+
+    private static PluginSettings INSTANCE;
+
+    private static final String KEYSTORE_FILENAME = "wazuh-indexer.keystore";
+
+    private static KeyStoreWrapper keyStoreWrapper = KeyStoreWrapper.create();
+    private static Environment env;
```
Comment (AlexRuiz7) [inline plugins/command-manager/src/main/java/com/wazuh/commandmanager/settings/PluginSettings.java:29]: Link these properties to the PluginSettings instance, not to the class. For that, remove their static declaration and initialize them where required:

- Environment -> `+PluginsSettings::setEnvironment()`
- KeyStoreWrapper -> `-PluginsSettings::PluginsSettings()` (constructor)

File: plugins/command-manager/src/main/java/com/wazuh/commandmanager/settings/PluginSettings.java [hunk 5]
```diff
@@ -0,0 +1,88 @@
...
+            INSTANCE = new PluginSettings();
+            return INSTANCE;
+        }
+    }
+
+    public void setEnv(Environment env) {
+        PluginSettings.env = env;
+    }
+
+    static SecureSettings loadSecureSettings(SecureString secureSettingsPassword) throws CommandManagerSettingsException, GeneralSecurityException {
```
Comment (AlexRuiz7) [inline plugins/command-manager/src/main/java/com/wazuh/commandmanager/settings/PluginSettings.java:52]: Static declaration is probably not required here. This business logic depends on the instance's state, and should be invoked as follows:

```java
PluginSettings settings = PluginSettings.getInstance();
settings.loadSecureSettings(password);
```

or, for short:

```java
PluginsSettings.getInstance().loadSecureSettings(password);
```

File: plugins/command-manager/src/main/java/com/wazuh/commandmanager/settings/PluginSettings.java [hunk 6]
```diff
@@ -0,0 +1,88 @@
...
+        try {
+            //Open the keystore file
+            keyStoreWrapper = KeyStoreWrapper.load(env.configFile(),KEYSTORE_FILENAME);
+            if (keyStoreWrapper == null) {
+                logger.info(CommandManagerSettingsException.keystoreNotExist(env.configFile().toString()).getMessage());
+
+                //Create keystore file if it doesn't exist
+                keyStoreWrapper = KeyStoreWrapper.create();
+                keyStoreWrapper.save(env.configFile(), new char[0]);
```
Comment (AlexRuiz7) [inline plugins/command-manager/src/main/java/com/wazuh/commandmanager/settings/PluginSettings.java:61]: Refer to class members using `this`.

File: plugins/command-manager/src/test/resources/wazuh-indexer.keystore.json [hunk 7]
```diff
@@ -0,0 +1,11 @@
+[
+  {
+    "name" : "wazuh-api",
+    "properties" : {
+      "prometheus.uri" : "http://localhost:9090",
+      "prometheus.auth.type" : "basicauth",
+      "prometheus.auth.username" : "admin",
+      "prometheus.auth.password" : "type"
```
Comment (AlexRuiz7) [inline plugins/command-manager/src/test/resources/wazuh-indexer.keystore.json:8]: Simplify to

```
      "uri" : "http://localhost:9090",
      "auth.type" : "basicauth",
      "auth.username" : "admin",
      "auth.password" : "type"
```

File: plugins/command-manager/src/main/java/com/wazuh/commandmanager/settings/CommandManagerSettings.java
Comment (AlexRuiz7) [inline plugins/command-manager/src/main/java/com/wazuh/commandmanager/settings/CommandManagerSettings.java:1]: This class should define these settings:

- URI
- AUTH_TYPE
- AUTH_USERNAME
- AUTH_PASSWORD

These relate to these fields in the key store:

```
      "uri" : "http://localhost:9090",
      "auth.type" : "basicauth",
      "auth.username" : "admin",
      "auth.password" : "type"
```

File: plugins/command-manager/build.gradle [hunk 8]
```diff
@@ -129,6 +129,9 @@ testClusters.integTest {
 
     // This installs our plugin into the testClusters
     plugin(project.tasks.bundlePlugin.archiveFile)
+
+    // add customized keystore
+    keystore 'command.manager.access_key', new File("$projectDir/src/test/resources/", 'wazuh-indexer.keystore.json')
```
Comment (AlexRuiz7) [inline plugins/command-manager/build.gradle:134]: Add these fields

```
      "uri" : "http://localhost:9090",
      "auth.type" : "basicauth",
      "auth.username" : "admin",
      "auth.password" : "type"
```

File: plugins/command-manager/src/main/java/com/wazuh/commandmanager/CommandManagerSettingsException.java
Comment (AlexRuiz7) [inline plugins/command-manager/src/main/java/com/wazuh/commandmanager/CommandManagerSettingsException.java:1]: Maybe we can move this class into the `settings` package.

**Comment by mcasas993** [comment]
###Opensearch solution
We ask for help to OpenSearch for managing the Settings in the Keystore, and they give us a simple solution:

The plugins can declare "secure" settings and then they will have access to the decrypted value directly. They only have access to their own declared secure settings, but not secure settings declared by other plugins

They can be declared like [this](https://github.com/opensearch-project/security/blob/main/src/main/java/org/opensearch/security/ssl/OpenSearchSecuritySSLPlugin.java#L409) and they can be accessed like [any other setting](https://github.com/opensearch-project/security/blob/main/src/main/java/org/opensearch/security/ssl/DefaultSecurityKeyStore.java#L601). 

The different between secureSetting vs normal setting is that secureSetting is stored in the keystore.

So we are going to refactor the code to simplify the solution.

File: plugins/command-manager/build.gradle [hunk 9]
```diff
@@ -185,10 +185,9 @@ testClusters.integTest {
     plugin(project.tasks.bundlePlugin.archiveFile)
 
     // add customized keystore
-    keystore 'command.manager.keystore', new File("$projectDir/src/test/resources/", 'wazuh-indexer.keystore.json')
     keystore 'command.manager.auth.username', 'admin'
     keystore 'command.manager.auth.password', 'test'
-
+    keystore 'command.manager.uri', 'https://httpbin.org/post'
```
[the remark this answers is not in the document]
Comment (mcasas993) [reply plugins/command-manager/build.gradle]: I can't use '_' and '.' in the name, one or another.

File: plugins/command-manager/src/test/resources/wazuh-indexer.keystore.json
Comment (AlexRuiz7) [inline plugins/command-manager/src/test/resources/wazuh-indexer.keystore.json:1]: Is this file in use?

Comment (mcasas993) [reply plugins/command-manager/src/test/resources/wazuh-indexer.keystore.json:1]: Finally, no.

**Pull request closed without merging**